gradle.properties

```diff
@@ -49,10 +52,20 @@ ekirjasto.languages=en,fi,sv
 ekirjasto.liblcp.uri=https://liblcp.dita.digital
 ekirjasto.liblcp.repositorylayout=/[organisation]/[module]/android/aar/test/[revision].[ext]
 
+# Override these in local.properties, not directly here
+ekirjasto.storePassword=storePassword
+ekirjasto.keyPassword=keyPassword
+ekirjasto.keyAlias=key0
+
 # Email address to send error reports to (the feedback form is handled differently)
 ekirjasto.supportEmail=e-kirjasto-tekniikka@helsinki.fi
```

Comment:
Are you suggesting that this would be moved to the backend and fetched on startup?

Reply:
I understand that these values will be used for reviewing the apps. I'm wondering if it is safe to have them in the clear, other values from the same file are either in GitHub secrets or overridden in the file local.properties. Also a new PR would have to be done for every modification (which might not happen often, but we had to reset the whole DB once and the collection id was changed). Of course we don't have much time either.

Reply:
The way this email is used is that when an error happens, this just opens your device's email app with the address given here, so the email is user-visible anyway. And before it was defined in this config file, it was hard-coded in a Kotlin file, so it was public through there. So this just moves a configuration value to a configuration file instead of having it directly inside code.

And the test login backend and library provider ID are also visible in simplified-app-ekirjasto's build.gradlle.kts, so those are also already public.

gradle.properties is really just a shared configuration file, it's not used for secrets at all. These are essentially default values that can be overridden in local.properties.

local.properties is a similar configuration file, but it's user-specific instead of shared, so it's not included in the repo, so it's also used for secrets. But it's not *just* for secrets, you can also use it e.g. to override the feedback form URL, which Max uses to develop it (by using a locally hosted feedback form).

Reply:
Continuing about the library provider IDs, here's the build file where they are normally defined (for each flavor):
https://github.com/NatLibFi/ekirjasto-android-core/blob/main/simplified-app-ekirjasto/build.gradle.kts#L190

And here's the same for iOS (this is for production, each environment has it's own file, and you have to replace the file instead of e.g. changing a build target, the build system isn't as nice):
https://github.com/NatLibFi/ekirjasto-ios-core/blob/main/Palace/EkirjastoConfig/Ekirjasto_Catalog_Feed.json#L44

So the way the Palace apps work, the library provider IDs have to be set like this on the client side. It would take some effort (also on the backend) to fetch these dynamically at app startup instead of defining them at compile-time like this.

Reply:
And about the email address, if you're worried about a bot picking it up from the GitHub repo (you mentioned spam), I could add some placeholder email here and override this value in local.properties through GitHub Actions secrets.

But then if we ever have to manually make a production release (not through GitHub Actions), there's a chance that whoever is making the build forgets to override the email through local.properties, so then the value here is used instead (so error reports would go the the wrong email for that manual release). Not sure if that's a big risk though, so either way is fine for me.

Reply:
Or, yet another option, we could for example base64 encode the email address here, and then decode it in the code. I thing that would be enough obfuscation for 99.9 % of the email spam bots out there.

Reply:
But anyway, none of these lines are even changed in this PR, so maybe any changes we want to make here could be made in another PR. This PR is already pretty big, I'd like to have minimal changes to the app itself here.